### PR TITLE
[IDE support] Peek and go to definition functionalities via jsconfig

### PR DESCRIPTION
**Pull request by sodic**
The implementation for Peek definition, go to definition, and import resolution in external JavaScript source files.

Takes steps towards fixing #710, and (more generally) #604 and #688.

### CODE REVIEW

**Comment by Martinsos** [comment]
Hm this PR @sodic has nothing but e2e updates! Is that a mistake?

**Comment by sodic** [comment]
It's a mistake, but not on my part :smile: 

It has changes in the `waspc/data` directory, but there are no code changes, that's correct.

**Comment by Martinsos** [comment]
Interesting, when I checked there were no changes under wasp/data. Maybe I followed the link that showed only the changes from the commit that updated e2e tests -> might be that.

File: waspc/data/Cli/templates/new/src/client/jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Martinsos) [inline waspc/data/Cli/templates/new/src/client/jsconfig.json:1]: Is there some stuff here that we don't want the users to touch?
Should we document it somehow, if so? Maybe directly in this file? I know JSON by itself doesn't support comments but there are tricks to add them.

File: waspc/data/Cli/templates/new/src/shared/jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "allowJs": true,
+    "target": "es2018",
+    "baseUrl": ".",
+    "paths": {
+      "*": [
+        "*",
+        "../../.wasp/out/server/node_modules/*",
```
Comment (Martinsos) [inline waspc/data/Cli/templates/new/src/shared/jsconfig.json:10]: Ok, we decided to go with server.
Do we have any idea how to make this "right" in the future? Well, as "right" as possible?

**Review by Martinsos — APPROVED** [review]
LGTM! Approved but also asked couple of questions.

File: waspc/data/Cli/templates/new/src/shared/jsconfig.json (hunk 2 above)
Comment (sodic) [reply waspc/data/Cli/templates/new/src/shared/jsconfig.json:10]: I can't think of a solution that doesn't involve separating server, client, and shared npm dependencies.

File: waspc/data/Cli/templates/new/src/client/jsconfig.json (hunk 1 above)
Comment (sodic) [reply waspc/data/Cli/templates/new/src/client/jsconfig.json:1]: Yes, definitely, I'll put it in.

**Pull request merged**